cpp/tests/merge/merge-test.cu

```diff
@@ -0,0 +1,378 @@
+#include <cassert>
+#include <vector>
+#include <memory>
+#include <algorithm>
+#include <limits>
+#include <gtest/gtest.h>
+#include <nvstrings/NVCategory.h>
+#include <nvstrings/NVStrings.h>
+
+#include <cudf/cudf.h>
+#include <cudf/functions.h>
+#include <cudf/merge.hpp>
+#include <rmm/thrust_rmm_allocator.h>
+#include <cudf/table.hpp>
+
+#include "string/nvcategory_util.hpp"
+#include "tests/utilities/column_wrapper.cuh"
+#include "tests/utilities/column_wrapper_factory.hpp"
+#include "tests/utilities/cudf_test_fixtures.h"
+#include "tests/utilities/nvcategory_utils.cuh"
+
+template <typename T>
+class MergeTest : public GdfTest {};
+
+using test_types =
+  ::testing::Types<int8_t, int16_t, int32_t, int64_t, float, double,
+                   /* cudf::bool8, */ cudf::nvstring_category>;
```

Comment:
Why is cudf::bool8 commented out?

Reply:
Am I correct that you still aren't testing anything with rows bigger than a single thread block can handle?

Reply:
I commented it out because of the conversion `0 => false any other value => true` doesn't map nicely to the labmda functions given to the `expectedData` column_wrappers. I could specific unit test for `cudf::bool8` if needed.
Already pushed a3e517174ed83a94014452d0e12b2a64751bf41a which updates the size of the test data.

Reply:
I think an easy change would be to check the source data column's dtype and if it is GDF_BOOL8, then change the lambda for the input data and the ExpectedData* to do something more interesting for bools, like return `row % 2` instead of `row`.

Reply:
Updated the unit test for `cudf::bool8` though they are not that interesting because the input data must be sorted so its just a bunch of 0's followed by 1's or vice versa

Reply:
That's OK. The point is to exercise that pathway and ensure there are no surprises.